### ISSUE TITLE
iOS build failed with 'exception' file not found

### ISSUE THREAD

**Issue by uragirii** [customerio/customerio-reactnative#105]
**SDK version:** 
Customerio-reactnative: v2.1.0
React Native: 0.64.4

**Environment:** Development 

**Describe the bug**
When I try to build the app using Xcode after installing package and doing pod install. I see this error in Xcode.
<img width="271" alt="Screenshot 2023-02-28 at 5 12 21 PM" src="https://user-images.githubusercontent.com/19703138/221843965-23a74895-6d36-40e2-92fe-9004d2324627.png">

**Expected behavior**
Apps should build successfully 

**Additional context**
Podfile:
```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '13.0'

target 'HybridApp' do
    use_unimodules!({
    modules_paths: ['../../../node_modules'],
    exclude: ['expo-web-browser', 'expo-crypto', 'expo-font']
  })
  config = use_native_modules!

  use_react_native!(
      :path => config["reactNativePath"],
      :hermes_enabled => true
  )

  if !ENV['CI']
    use_flipper!({ 'Flipper' => '0.140.0', 'Flipper-Folly' => '2.6.10' })
    post_install do |installer|
      react_native_post_install(installer)
    end
  end
end
```

**Comment by Shahroz16** [comment]
Hey @uragirii, thank you for reaching out. It's a bit hard to know what might be going wrong with the given information and the error point to setup/podfile issues. 

Can you please look into our [reference app](https://github.com/customerio/amiapp-reactnative) and [documentation](https://www.customer.io/docs/sdk/react-native/getting-started/) to make sure the setup is correct?

**Comment by uragirii** [comment]
Hi @Shahroz16, yes followed all the steps as per your documentation. 

I tried to scaffold a new RN app with cli also (`npx react-native@0.64.4 init CioTest --version=0.64.4`) and then followed all the steps as per your documentation but i still got the same error. This makes me to believe that there is something wrong with this version of react native. 

I also find [this comment](https://github.com/lottie-react-native/lottie-react-native/issues/708#issuecomment-775175081) regarding same problem but with `lottie-react-native` for the same react native version (0.64). I'm not familiar with iOS development, can you look at this and see if anything makes sense?

FWIW, i tried to run app with cli (instead of xcode) and got this error : 
```
The following build commands failed:
        SwiftEmitModule normal x86_64 Emitting\ module\ for\ customerio_reactnative (in target 'customerio-reactnative' from project 'Pods')
(1 failure)```

<img width="1408" alt="Screenshot 2023-03-02 at 1 10 59 AM" src="https://user-images.githubusercontent.com/19703138/222246984-0c6992be-5c55-45ae-a5ba-44033e1832f4.png">

**Comment by Shahroz16** [comment]
Hey @uragirii, apologies that this one slipped through the notifications. Can you please share the test app your created with CIO integrated? We can probably look into that and see what might be the problem. 

Meanwhile, can you try following, follow the [pod file](https://github.com/customerio/amiapp-reactnative/blob/main/ios/Podfile) present in your sample app? 

After that, you can also try the following : 

```
cd ios
rm -rf Pods
rm -rf Podfile.lock
pod install
cd ..
react-native run-ios
```

**Comment by uragirii** [comment]
No worries! Thanks for the reply. I tried using your Pod file but i get weird errors. Maybe its not compatible. I created the repo for your reference https://github.com/uragirii/ciotest

**Comment by ami-aman** [comment]
Hi @uragirii 
Jumping in for Shahroz here. First of all, I appreciate your patience.

I checked the [Github sample project](https://github.com/uragirii/ciotest) and was able to reproduce the errors you mentioned. This appears to be related to react native version 0.64.4 and there is a similar open [ticket](https://github.com/facebook/react-native/issues/35601) with react native.

I was able to resolve the problem by updating the react native version to `0.65.3`, which includes the default Hermes engine version of `0.8.1`. Use this [react native upgrade helper](https://react-native-community.github.io/upgrade-helper/) to upgrade to a version that has the fix for this.

After upgrading the version, I recommend uninstalling node_modules and pods and reinstalling them. To accomplish the same thing, use the commands listed below.
- rm -rf node_modules && rm -rf ios/Pods
- yarn install
- cd ios && pod install && cd ..

Replace yarn with npm if required.

Let me know if this fixed the error for you.
